images/ceph/Makefile

```diff
@@ -68,33 +70,36 @@ export OPERATOR_SDK YQv3
 
 do.build:
 	@echo === container build $(CEPH_IMAGE)
-	@cp Dockerfile $(TEMP)
-	@cp toolbox.sh $(TEMP)
-	@cp set-ceph-debug-level $(TEMP)
-	@cp $(OUTPUT_DIR)/bin/linux_$(GOARCH)/rook $(TEMP)
-	@cp -r $(MANIFESTS_DIR)/monitoring $(TEMP)/ceph-monitoring
-	@mkdir -p $(TEMP)/rook-external/test-data
-	@cp $(MANIFESTS_DIR)/create-external-cluster-resources.* $(TEMP)/rook-external/
-	@cp ../../tests/ceph-status-out $(TEMP)/rook-external/test-data/
+	@mkdir -p $(BUILD_CONTEXT_DIR)
+	@cp Dockerfile $(BUILD_CONTEXT_DIR)
+	@cp toolbox.sh $(BUILD_CONTEXT_DIR)
+	@cp set-ceph-debug-level $(BUILD_CONTEXT_DIR)
+	@cp $(OUTPUT_DIR)/bin/linux_$(GOARCH)/rook $(BUILD_CONTEXT_DIR)
+	@cp -r $(MANIFESTS_DIR)/monitoring $(BUILD_CONTEXT_DIR)/ceph-monitoring
+	@mkdir -p $(BUILD_CONTEXT_DIR)/rook-external/test-data
+	@cp $(MANIFESTS_DIR)/create-external-cluster-resources.* $(BUILD_CONTEXT_DIR)/rook-external/
+	@cp ../../tests/ceph-status-out $(BUILD_CONTEXT_DIR)/rook-external/test-data/
 
 ifeq ($(INCLUDE_CSV_TEMPLATES),true)
 	@$(MAKE) csv
-	@cp -r ../../build/csv $(TEMP)/ceph-csv-templates
-	@rm $(TEMP)/ceph-csv-templates/csv-gen.sh
+	@cp -r ../../build/csv $(BUILD_CONTEXT_DIR)/ceph-csv-templates
+	@rm $(BUILD_CONTEXT_DIR)/ceph-csv-templates/csv-gen.sh
 	@$(MAKE) csv-clean
 
 else
-	mkdir $(TEMP)/ceph-csv-templates
+	mkdir $(BUILD_CONTEXT_DIR)/ceph-csv-templates
 endif
-	@cd $(TEMP) && $(SED_IN_PLACE) 's|BASEIMAGE|$(BASEIMAGE)|g' Dockerfile
+	@cd $(BUILD_CONTEXT_DIR) && $(SED_IN_PLACE) 's|BASEIMAGE|$(BASEIMAGE)|g' Dockerfile
 	@if [ -z "$(BUILD_CONTAINER_IMAGE)" ]; then\
 		$(DOCKERCMD) build $(BUILD_ARGS) \
 		--build-arg S5CMD_VERSION=$(S5CMD_VERSION) \
 		--build-arg S5CMD_ARCH=$(S5CMD_ARCH) \
 		-t $(CEPH_IMAGE) \
-		$(TEMP);\
+		$(BUILD_CONTEXT_DIR);\
+	fi
+	@if [ -z "$(SAVE_BUILD_CONTEXT_DIR)" ]; then\
+		rm -fr $(BUILD_CONTEXT_DIR);\
 	fi
-	@rm -fr $(TEMP)
```

Comment:
Without this line, users that don't use the BUILD_CONTEXT_DIR will have loads of temp files that aren't getting cleaned up. Please ensure somehow that this dir is cleaned up in the default case.

Reply:
I'm not sure I understand how the cleanup is not called in the default case. Cleaning up in the default case still happens with the change in this PR.

`BUILD_CONTEXT_DIR` is set if a user does not set it here:
```
ifeq ($(BUILD_CONTEXT_DIR),)
BUILD_CONTEXT_DIR := $(shell mktemp -d)
endif
```

`BUILD_CONTEXT_DIR` is removed if the `SAVE_BUILD_CONTEXT_DIR` variable is not set:

```
@if [ -z "$(SAVE_BUILD_CONTEXT_DIR)" ]; then\
	rm -fr $(BUILD_CONTEXT_DIR);\
fi
```

Reply:
Oh. I must have missed that. That was my only concern, and I see that I just missed the deail, so looks good to me